packages/flutter/lib/src/material/carousel.dart

```diff
@@ -448,6 +452,46 @@ class CarouselView extends StatefulWidget {
   /// The child widgets for the carousel.
   final List<Widget> children;
 
+  /// {@template flutter.material.CarouselView.onIndexChanged}
+  /// A callback invoked when the leading item changes.
+  ///
+  /// The “leading” item is the one that the carousel resolves as primary for
+  /// the current frame according to its layout algorithm. This item can be only
+  /// partially visible while scrolling.
+  ///
+  /// - In a standard [CarouselView], the leading item is the one positioned at
+  ///   the leading edge of the viewport based on the current scroll offset.
+  ///
+  /// - In a [CarouselView.weighted], the leading item is chosen by the weighted
```

Comment:
The leading item is also the first (partially visible) item for `CarouselView.weighted`. What I meant to say in this [comment](https://github.com/flutter/flutter/pull/169271#discussion_r2579647163) is that we don't need to distinguish the definitions of the 2 leading items now. They both mean the first item in the view. Please let me know if my understanding is wrong.